htdocs/core/lib/files.lib.php

```diff
@@ -2799,20 +2799,19 @@ function dol_check_secure_access_document($modulepart, $original_file, $entity,
 				exit;
 			}
 
-			/*$perm = GETPOST('perm', 'aZ09');
-			 $subperm = GETPOST('subperm', 'aZ09');
-			 if ($perm || $subperm)
-			 {
-			 if (($perm && !$subperm && $fuser->rights->$modulepart->$perm) || ($perm && $subperm && $fuser->rights->$modulepart->$perm->$subperm)) $accessallowed = 1;
-			 }
-			 else
-			 {*/
-			// Check fuser->rights->modulepart->myobject->read and fuser->rights->modulepart->read
-			$partsofdirinoriginalfile = explode('/', $original_file);
-			$partofdirinoriginalfile = $partsofdirinoriginalfile[0];
-			if ($partofdirinoriginalfile && ($fuser->rights->$modulepart->$partofdirinoriginalfile->{$lire} || $fuser->rights->$modulepart->$partofdirinoriginalfile->{$read})) $accessallowed = 1;
-			if ($fuser->rights->$modulepart->{$lire} || $fuser->rights->$modulepart->{$read}) $accessallowed = 1;
-			//}
+			$perm = GETPOST('perm', 'aZ09');
```

Comment:
Testing permissions of a field that is provided by the user is useless. Any user can set the value of its choice to gain access. Why do you want to add this ? For external module ? If so only modulepart must be provided. Check on permission must be based on the same parameter than the parameter to decide the directory we want access to.

Reply:
Exactly, it is for external modules that doesn't use `lire` or `read` permissions by default, for example: `access` or `see` or whatever permission name. I know that a simple solution would be to rename the permissions to read, but let's assume that we have a subpermission case, using only read or lire perm will not help in that case since we cannot even define the subpermission, so, is there anyway to make/add a condition to let that work only for external modules? (because specific permissions are already set for most of core modules)

Reply:
> Any user can set the value of its choice to gain access.

I think that even if the user set the value of his choice (using url parameters), he still need to have that permission, if he doesn't he wont be able to access.